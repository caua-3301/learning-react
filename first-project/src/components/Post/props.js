export const props = {
  posts: [
    {
      id: 1,
      photoUrl: 'foto1.png',
      title: 'titulo1',
      completed: true,
    },

    {
      id: 2,
      photoUrl: 'foto2.png',
      title: 'titulo2',
      completed: false,
    },

    {
      id: 3,
      photoUrl: 'foto3.png',
      title: 'titulo3',
      completed: true,
    },
  ],
};
