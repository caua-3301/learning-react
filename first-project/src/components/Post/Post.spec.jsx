import userEvent from "@testing-library/user-event";
import { screen, render } from "@testing-library/react";
import { Posts } from ".";
import { props } from "./props";

describe("<Post/>", () => {
    it("should exibe all imgs", () => {
        //props representa o props recebido la no arquivo index.jsx "export const Posts = (posts = []) =>"
        render(<Posts {...props} />);

        const allImgs = screen.getAllByRole("img");

        //verificando atributos
        expect(allImgs[0]).toHaveAttribute("src", "foto1.png");
        //checando renderização

        allImgs.forEach(image => {
            expect(image).toBeVisible();
            expect(image).toHaveAttribute('src');
        });

        expect(allImgs).toHaveLength(3);
    });

    it("should make the snapshot", () => {
        //props representa o props recebido la no arquivo index.jsx "export const Posts = (posts = []) =>"
        render(<Posts {...props} />);

        const elementProp = document.body;

        expect(elementProp).toMatchSnapshot();
    })
});