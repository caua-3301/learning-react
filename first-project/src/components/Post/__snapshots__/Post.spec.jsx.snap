// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post/> should make the snapshot 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div
        class="card"
      >
        <img
          src="foto1.png"
        />
        <h4>
          Titulo: 
          titulo1
           
        </h4>
        <p>
           
          Conpletada
           
        </p>
      </div>
      <div
        class="card"
      >
        <img
          src="foto2.png"
        />
        <h4>
          Titulo: 
          titulo2
           
        </h4>
        <p>
           
          Por fazer
           
        </p>
      </div>
      <div
        class="card"
      >
        <img
          src="foto3.png"
        />
        <h4>
          Titulo: 
          titulo3
           
        </h4>
        <p>
           
          Conpletada
           
        </p>
      </div>
    </div>
  </div>
</body>
`;
